src/test/java/com/amazonaws/encryptionsdk/kms/AwsKmsMrkAwareMasterKeyProviderTest.java

```diff
@@ -225,33 +224,23 @@ public void no_duplicate_mrks() {
 
     @Test
     public void always_need_a_region() {
-      assertThrows(
-          AwsCryptoException.class,
-          () ->
-              AwsKmsMrkAwareMasterKeyProvider.builder()
-                  .withDefaultRegion(null)
-                  .buildStrict("mrk-edb7fe6942894d32ac46dbb1c922d574"));
-
       AwsKmsMrkAwareMasterKeyProvider.builder()
-          .withDefaultRegion("us-east-1")
+          // If default region is set to `null` or not configured
+          // it will pick one up from the environment
+          .withDefaultRegion(null)
           .buildStrict("mrk-edb7fe6942894d32ac46dbb1c922d574");
     }
 
     @Test
-    // = compliance/framework/aws-kms/aws-kms-mrk-aware-master-key-provider.txt#2.6
-    // = type=test
-    // # If an AWS SDK Default Region can not be
-    // # obtained initialization MUST fail.
```

Comment:
The check is present because `KmsMasterKeyProvider` allowed `_defaultregion` to be set to null during the build. This is no longer the case with this change. By deferring the call to `getSdkDefaultRegion()` unless no default region is provided, `_defaultregion` will always be a non-null value unless `DefaultAwsCredentialsProvider` could not fetch the region.

Reply:
Wait... 
```
    private static String getSdkDefaultRegion() {
      try {
        return new com.amazonaws.regions.DefaultAwsRegionProviderChain().getRegion();
      } catch (SdkClientException ex) {
        return null;
      }
    }
```
This method literally returns `null` in case of an Exception. 
I think this introduces an NPE...

Reply:
No, b/c it was already set to be this way.
Darwin has just moved the call further down the process, as the requester has asked.